lib/spack/spack/test/directives.py

```diff
@@ -89,6 +89,12 @@ def test_maintainer_directive(config, mock_packages, package_name, expected_main
     assert pkg_cls.maintainers == expected_maintainers
 
 
+@pytest.mark.parametrize("package_name,expected_licenses", [("licenses-1", ["MIT", "Apache-2.0"])])
+def test_license_directive(config, mock_packages, package_name, expected_licenses):
+    pkg_cls = spack.repo.PATH.get_pkg_class(package_name)
+    assert expected_licenses == list(pkg_cls.licenses.keys())
```

Comment:
Let's also add a check here for `when=` to test it's being set as we'd expect. Otherwise this is looking really good.